deal_thief/conftest.py

```diff
@@ -0,0 +1,106 @@
+"""Conftest file the does the prep work prior tests run."""
+import pytest
+from passlib.apps import custom_app_context as pwd_context
+from pyramid import testing
+from .models import (
+        User,
+        Search,
+        get_engine,
+        get_session_factory,
+        get_tm_session
+)
+from .models.meta import Base
+import transaction
+
+
+class MockRequest(object):
+    """MockRequest intended to be a better version of testing.DummyRequest."""
+
+    def __init__(self, **kwargs):
+        """Init a MockRequest instance."""
+        self.__dict__.update(kwargs)
+
+    def route_url(self, string):
+        """Mimic the pyramid.request.Request route_url."""
+        return '/{}'.format(string)
+
+
+class MockSession(object):
+    def __init__(self, **kwargs):
+        """Init a MockSession instance."""
+        self.flash_msg = []
+        self.__dict__.update(kwargs)
+
+    def flash(self, string):
+        """Mimic the pyramid.session.flash object."""
+        self.flash_msg.append(string)
```

Comment:
Interesting.  I would have inherited from the base `dict` type if I'd done this, so that you get the dictionary-like behavior of the session for free.  But I'm not convinced even that is required.